x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/LoadMapping.java

```diff
@@ -108,6 +111,9 @@ private static void walkMapping(String name, Object value, Map<String, EsField>
                 field = new KeywordEsField(name, properties, docValues, length, normalized);
             } else if (esDataType == DATETIME) {
                 field = DateEsField.dateEsField(name, properties, docValues);
+            } else if (esDataType == SEMANTIC_TEXT) {
+                String inferenceId = textSetting(content.get("inference_id"), "test_inference_id");
+                field = new SemanticTextEsField(name, properties, docValues, false, Set.of(inferenceId));
```

Comment:
> > SemanticTextEsField is expected to have multiple inferenceIds according to its definition, we only load one here. 

This is only used in tests - and AFAIU this is loading a single mapping, so loading a single inference ID is expected.
When defining a `semantic_text` in a mapping for a single index, only one `inference_id` can be used.
`SemanticTextEsField` has more than one, because we could be querying more than one index and the `inference_ids` can be different.
The fixture I used in for `semantic_text` is only using `inference_id`.
You are right that there is also a `search_inference_id`, from the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/semantic-text.html#semantic-text-params):
```

PUT my-index-000002
{
  "mappings": {
    "properties": {
      "inference_field": {
        "type": "semantic_text",
        "inference_id": "my-elser-endpoint-for-ingest",
        "search_inference_id": "my-elser-endpoint-for-search"
      }
    }
  }
}
```

>  and I wonder which one do we use in ES|QL?

In ES|QL we only care about querying - so we should use `search_inference_id` - but `search_inference_id` is optional and defaults to `inference_id`. I will change the naming in `SemanticTextEsField` from `inferenceIds` to  `searchInferenceIds` to make it more obvious we are dealing with `search_inference_id`.